client-spark/shuffle-manager-3/src/main/java/org/apache/spark/shuffle/rss/HashBasedShuffleWriter.java

```diff
@@ -150,12 +160,21 @@ public HashBasedShuffleWriter(
             rssShuffleClient,
             writeMetrics::incBytesWritten,
             mapStatusLengths);
+
+    if (RssConf.columnarShuffleEnabled(rssConf)) {
+      this.schema = SparkUtils.getShuffleDependencySchema(dep);
+      this.isColumnarShuffle = RssColumnarBatchBuilder.supportsColumnarType(schema);
```

Comment:
RssColumnarBatchBuilder Initialization is only required if this condition is true